### DISCUSSION TITLE
Route JSON format (offline)

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#337] [Q&A] [answered]
I wrote the route JSON format many years ago, so Cruiser could store calculated routes offline.
It is an offline format that saves everything for navigation, it can be used offline without internet.

A few years later, with another extension (and some extra tags), I offered it in the curvy apps.

---

We can restore the JSON format option in the export & import of Cruiser.
It is already used in [favorite routes](https://github.com/devemux86/cruiser/discussions/35) and will help with many new features.

It is similar, but can include more data, with:
- https://github.com/devemux86/cruiser/discussions/250

**Comment by 0709wiwiwi** [comment]
Good morning Emux.
The announcement seems promising nice, but doesn't a unique Cruiser .json format not to lead a lonely life only useful in Cruiser? 
What is missing in gpx provided you do use the Cruiser extension inventively ?
Anyway the gpx advantage remains to be attractive for basic interchanges with other apps, even when apps lacks the niceties been added into the extra data contained into the Cruiser gpx extension.

**Reply by devemux86** [reply]
GPX always remains the best format for communication between apps or devices.

We can export the same route in GPX and JSON and compare the included data.
e.g. encoded coordinates, waypoints, speed limits, road information, ...

**Comment by devemux86** [comment, marked as answer]
Route JSON (offline) format is available in [Cruiser 3.0.34](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

**Comment by airald** [comment]
This format is very interesting because it saves a route taken with the parametes the user setted up when create it. It's very usefull to share with other route mates.

Is a open format that other apps could use or is just for cruiser?

**Reply by devemux86** [reply]
Json is always an open format and it is used in online routers in _many forms_.

Online routers generally prefer to send their route responses in json format,
because it is smaller, it can be read faster and can store extra custom data.
